mythril/interfaces/cli.py

```diff
@@ -300,55 +325,57 @@ def main():
 
     # -- commands --
     if args.hash:
-        print(Mythril.hash_for_function_signature(args.hash))
+        print(MythrilDisassembler.hash_for_function_signature(args.hash))
         sys.exit()
 
     try:
         # the mythril object should be our main interface
         # infura = None, rpc = None, rpctls = None
         # solc_args = None, dynld = None, max_recursion_depth = 12):
-
-        mythril = Mythril(
-            solv=args.solv,
-            dynld=args.dynld,
-            onchain_storage_access=(not args.no_onchain_storage_access),
-            solc_args=args.solc_args,
-            enable_online_lookup=args.query_signature,
-        )
+        config = MythrilConfig()
         if (
             args.dynld
             or not args.no_onchain_storage_access
             and not (args.rpc or args.i)
         ):
-            mythril.set_api_from_config_path()
+            config.set_api_from_config_path()
 
         if args.address:
             # Establish RPC connection if necessary
-            mythril.set_api_rpc(rpc=args.rpc, rpctls=args.rpctls)
+            config.set_api_rpc(rpc=args.rpc, rpctls=args.rpctls)
         elif args.search or args.contract_hash_to_address:
             # Open LevelDB if necessary
-            mythril.set_api_leveldb(
-                mythril.leveldb_dir if not args.leveldb_dir else args.leveldb_dir
+            config.set_api_leveldb(
+                config.leveldb_dir if not args.leveldb_dir else args.leveldb_dir
             )
 
-        if args.search:
-            # Database search ops
-            mythril.search_db(args.search)
-            sys.exit()
+        if args.search or args.contract_hash_to_address:
+            leveldb_searcher = MythrilLevelDB(config.eth_db)
+            if args.search:
+                # Database search ops
+                leveldb_searcher.search_db(args.search)
 
-        if args.contract_hash_to_address:
-            # search corresponding address
-            try:
-                mythril.contract_hash_to_address(args.contract_hash_to_address)
-            except AddressNotFoundError:
-                print("Address not found.")
+            else:
+                # search corresponding address
+                try:
+                    leveldb_searcher.contract_hash_to_address(
+                        args.contract_hash_to_address
+                    )
+                except AddressNotFoundError:
+                    print("Address not found.")
 
             sys.exit()
 
+        dissasembler = MythrilDisassembler(
+            eth=config.eth,
+            solc_version=args.solv,
+            solc_args=args.solc_args,
+            enable_online_lookup=args.query_signature,
+        )
         if args.truffle:
             try:
                 # not really pythonic atm. needs refactoring
-                mythril.analyze_truffle_project(args)
+                dissasembler.analyze_truffle_project(args)
```

Comment:
I'd put this in MythrilAnalyzer (but since truffle stuff is being depricated I dont mind ^^')